pkg/services/ngalert/state/manager.go

```diff
@@ -300,6 +300,10 @@ func (st *Manager) staleResultsHandler(ctx context.Context, alertRule *ngModels.
 			if err = st.instanceStore.DeleteAlertInstance(ctx, s.OrgID, s.AlertRuleUID, labelsHash); err != nil {
 				st.log.Error("unable to delete stale instance from database", "error", err.Error(), "orgID", s.OrgID, "alertRuleUID", s.AlertRuleUID, "cacheID", s.CacheId)
 			}
+
+			if s.State == eval.Alerting {
+				st.annotateState(ctx, alertRule, s.Labels, time.Now(), eval.Normal, s.State)
```

Comment:
I agree that we should put some state when a dimension disappears. However, I am not sure that Normal is the correct one. I wonder if it is possible to put "NoData" state with probably some description that mentions that evaluation result did not have that dimension.

Reply:
@yuri-tceretian, as @grobinson-grafana said [here](https://github.com/grafana/support-escalations/issues/1935#issuecomment-1048916161):

>The alert is resolved in Alertmanager if it doesn't receive an update from Grafana within 3x the evaluation interval. For example, if an alert rule is set to evaluate once per 10 seconds, and the series stops returning data for a minute, then 30 seconds after the last evaluation the alert will be resolved in Alertmanager.

So if the alert is being resolved in Alertmanager when the data stops, does it make sense to change the state to OK?

Tho I also understand your point that "OK" does not really capture what actually happened...

Reply:
Makes sense, I did not consider the fact that it is not deleted right away, only if it is considered as stale